### PR TITLE
Create a service used to map asset ip and asset name

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+         }else {
+            logger.error("Not exist asset with name : " + assetName);
+            return false;
+         }
+      }else {
+         Set<String> assetNames = getAllAssetNames(AssetCategory.Server);
+         if(assetNames.contains(mapping.getAssetname())) {
+            assetIPMappingRepository.save(mapping);
+         }else {
+            return false;
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java:59]: update the asset in the list.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+         }
+         redisService.opsForSetToAdd(SERVER_ASSET_NAME_LIST, SERVER_ASSET_NAME_TIME_OUT, assetNames.toArray(new String[assetNames.size()]));
+      }
+      return true;
+   }
+
+   public List<String> searchAssetName(String content){
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java:66]: searchServerAssetName

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            return allMatchResult.subList(0, LIMIT_RESULT + 1);
+         }
+         return allMatchResult;
+      }
+      Set<String> assetNames = getAllAssetNames(AssetCategory.Server);
+      redisService.opsForSetToAdd(SERVER_ASSET_NAME_LIST, SERVER_ASSET_NAME_TIME_OUT, assetNames.toArray(new String[assetNames.size()]));
+      if(assetNames.size() > LIMIT_RESULT) {
+         return new ArrayList<String>(assetNames).subList(0, LIMIT_RESULT + 1);
+      }
+      return new ArrayList<String>(assetNames);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java:82]: need filter the assetName before return.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+   }
+
+   private Set<String> getAllAssetNames(AssetCategory category){
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java:85]: just read from repo  directly.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java (hunk 1 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetIPAndNameMappingService.java]: done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 5]
```diff
@@ -125,6 +151,64 @@
       return result;
    }
 
+   public boolean isAssetNameValidate(AssetIPMapping mapping) {
+      if(redisTemplate.hasKey(SERVER_ASSET_NAME_LIST)) {
+         String assetName =  mapping.getAssetname();
+         if(!redisTemplate.opsForSet().isMember(SERVER_ASSET_NAME_LIST, assetName)) {
+            logger.error("Not found this item in redis : " + assetName);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:158]: info

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 6]
```diff
@@ -125,6 +151,64 @@
...
+         if(serverNames.size() < LIMIT_RESULT + 1) {
+            serverNames.add(assetName);
+         }
+      }
+      return serverNames;
+   }
+
+   private Set<String> getAssetNamesAndUpdateCache() {
+      List<Asset> assets = assetRepository.findAssetNameByCategory(AssetCategory.Server.name());
+      Set<String> assetNames = new LinkedHashSet<String>();
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:201]: Let's change to HashSet

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 7]
```diff
@@ -125,6 +151,64 @@
...
+      for(Asset asset : assets) {
+         assetNames.add(asset.getAssetName());
+      }
+      redisTemplate.opsForSet().add(SERVER_ASSET_NAME_LIST, assetNames.toArray(new String[assetNames.size()]));
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:205]: Let's make sure it overwrite.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 5 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 6 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 7 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:205]: Merge option,

**Pull request merged**